### ISSUE TITLE
permalinks for FB RSVPs do not have link to original

### ISSUE THREAD

**Issue by kylewm** [snarfed/bridgy#397]
all of these event RSVPs were sent to my event (the correct URL is there in the targets column)

![bridgy-events](https://cloud.githubusercontent.com/assets/950127/7519860/b47546e0-f497-11e4-9e23-7a884a08e076.png)

but the [RSVP permalinks](https://brid-gy.appspot.com/rsvp/facebook/12802152/868304959882663/10104096231146827) themselves have only u-in-reply-to the facebook URL. Since there is no link from the source to the target (my event post), they are rejected

```
  <a class="u-in-reply-to" href="https://www.facebook.com/events/868304959882663/"></a>
```

I'm sure this is related to the 2.0 conversion, but my intuition is failing me for *how*.

**Comment by snarfed** [comment]
:(:(:(:(:(

just let me know when you're ready to drop Facebook support, I'm totally on board

**Comment by kylewm** [comment]
How I wish you weren't kidding!  :smile: :book: :gun:

**Comment by snarfed** [comment]
emoji portmanteau!!! :heart:

**Comment by kylewm** [comment]
well it seems original_post_discovery is not finding the link in the post content because the content is just 'Kyle Mahan created an event.'

```
{'actor': {'displayName': u'Kyle Mahan',
           'id': u'tag:facebook.com,2013:12802152',
           'image': {'url': u'https://graph.facebook.com/v2.2/12802152/picture?type=large'},
           'numeric_id': u'12802152',
           'objectType': 'person',
           'url': u'https://www.facebook.com/12802152'},
 'fb_id': u'12802152_868304959882663',
 'id': u'tag:facebook.com,2013:868304959882663',
 'object': {'attachments': [{'content': u'Are you building your own website? Indie reader? Personal publishing web app? Or some other digital magic-cloud proxy? If so, come on by and join a gathering of people with like-minded interests. Bring your friends that want to start a personal web site. Exchange information, swap ideas, talk shop, help work on a project.\n\nMore information: https://indiewebcamp.com/events/2015-05-06-homebrew-website-club\n\nIndie-event: https://kylewm.com/2015/04/homebrew-website-club-2',
                             'displayName': u'Homebrew Website Club: 2015 May 6',
                             'image': {'url': u'https://scontent.xx.fbcdn.net/hphotos-xtf1/v/t1.0-9/c29.0.130.130/p130x130/11152357_10101288077224839_6205512896323225998_n.jpg?oh=9b54e36b1aacdcee3e250d6916caca4c&oe=55CA3C5B'},
                             'objectType': 'article',
                             'summary': u'Homebrew Website Club: 2015 May 6',
                             'url': u'https://www.facebook.com/events/868304959882663/'}],
            'author': {'displayName': u'Kyle Mahan',
                       'id': u'tag:facebook.com,2013:12802152',
                       'image': {'url': u'https://graph.facebook.com/v2.2/12802152/picture?type=large'},
                       'numeric_id': u'12802152',
                       'objectType': 'person',
                       'url': u'https://www.facebook.com/12802152'},
            'content': u'Kyle Mahan created an event.',
            'fb_id': u'12802152_868304959882663',
            'fb_object_id': u'868304959882663',
            'id': u'tag:facebook.com,2013:868304959882663',
            'image': {'url': u'https://scontent.xx.fbcdn.net/hphotos-xtf1/v/t1.0-9/c29.0.130.130/p130x130/11152357_10101288077224839_6205512896323225998_n.jpg?oh=9b54e36b1aacdcee3e250d6916caca4c&oe=55CA3C5B'},
            'objectType': 'event',
            'published': '2015-04-25T06:58:19+00:00',
            'to': [{'alias': '@public', 'objectType': 'group'}],
            'updated': '2015-04-25T06:58:19+00:00',
            'url': u'https://www.facebook.com/12802152/posts/868304959882663'},
 'published': '2015-04-25T06:58:19+00:00',
 'updated': '2015-04-25T06:58:19+00:00',
 'url': u'https://www.facebook.com/12802152/posts/868304959882663',
 'verb': 'post'}
```

**Comment by kylewm** [comment]
Seems to be the difference between 

12802152_868304959882663 which is the ID of the "Created an event" activity

and

868304959882663 which is the ID of the event.

We've been assuming that USERID_POSTID was always equivalent to POSTID (if both were available), but apparently they aren't, at least for events.

**Comment by snarfed** [comment]
yup, you beat me to it on the distinction between "created an event" post objects and the event objects themselves. so if i understand right, we [get event objects themselves ok in poll](https://github.com/snarfed/bridgy/blob/master/facebook.py#L114), but we fetch the post objects instead in the mf2 handler?

**Comment by kylewm** [comment]
Yep, exactly. I think it's as simple as switching around the order that we try the activity_ids in, so that we try the one we're given first before trying the constructed one. I'm playing with

```
        ids_to_try = ['_'.join((user_id, activity_id)), activity_id]
to
        ids_to_try = [activity_id, '_'.join((user_id, activity_id))]
```

**Comment by snarfed** [comment]
hmm. that's [here](https://github.com/snarfed/activitystreams-unofficial/blob/master/facebook.py#L205)? i think we did that until a few weeks ago, then changed it in 19ea8c02cd277996a47d45f60ad9b3f94dab2d94 due to #388. would that regress then?

**Comment by kylewm** [comment]
damn, yes and yes. will give it some more thought, let me know if you have any ideas!

**Comment by snarfed** [comment]
sounds like we'd want to special case [`Facebook.get_rsvp` in a-u](https://github.com/snarfed/activitystreams-unofficial/blob/master/facebook.py#L300)?

...although following the chain from [`RsvpHandler`](https://github.com/snarfed/bridgy/blob/master/handlers.py#L259) through there and beyond, i don't actually see that we fetch the event/post at all, just the rsvp...?

**Comment by kylewm** [comment]
`add_original_post_urls` fetches the post

**Comment by snarfed** [comment]
three ideas from ongoing irc discussion:
1. handle the more detailed activity we're given from `userid_postid`
1. move this half-assed event detection into a-u and do it right, so that this Just Works: https://github.com/snarfed/bridgy/blob/master/facebook.py#L120
1. pull the initial post fetch out of `add_original_post_urls` in handlers.py, do it for non-rsvps, and special case event objects for rsvps

**Comment by kylewm** [comment]
just following up our conversation from IRC

>    handle the more detailed activity we're given from userid_postid

I don't really like this idea anymore (even though I proposed it). The userid_postid is actually a different activity (the action of posting the event, rather than the event itself). It *happens* to have some of the info we need but it doesn't feel reliable.

>    move this half-assed event detection into a-u and do it right, so that this Just Works: https://github.com/snarfed/bridgy/blob/master/facebook.py#L120

I think this would be a good change, but it wouldn't solve this particular issue, afaict. The augmentation logic in bridgy is for adding events and photos to the full stream of posts. It wouldn't have an effect on fetching a single event by its id.

>    pull the initial post fetch out of add_original_post_urls in handlers.py, do it for non-rsvps, and special case event objects for rsvps

I think this is the most promising. If we added models.get_event and au_source.get_event, then Facebook could apply special logic for fetching events that it wouldn't use for the other posts. And I like your suggestion to supply the post-content optionally so the other cases can all stay the same.
